Engine/Component/ComponentScript.cpp

```diff
@@ -50,6 +50,7 @@ void ComponentScript::Update()
 	if (script && active) 
 	{
 		script->Update();
+		script->UpdateWithImGui(ImGui::GetCurrentContext());
```

Comment:
The idea is not to adapt the engine to an specific scripts and this is to specific, so brake the "encapsulation" we are trying to achieve. So instead of that, maybe you could make a function in the engine that returns you the current context. And then in the Update you can call that. and use that context.